Crashes under Windows

### Describe the bug

Adding this here so we don't forget to fix it

### Reproduction

https://github.com/vitest-dev/vitest/runs/4868391749?check_suite_focus=true

### System Info

```shell
Windows
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

nopes, still failing:
```shell
examples/mocks test:  1: 00007FF60F6E013F v8::internal::CodeObjectRegistry::~CodeObjectRegistry+112495
examples/mocks test:  2: 00007FF60F66F396 DSA_meth_get_flags+65526
examples/mocks test:  3: 00007FF60F67024D node::OnFatalError+301
examples/mocks test:  4: 00007FF60FFAA455 v8::V8::ToLocalEmpty+53
examples/mocks test:  5: 00007FF60F667208 DSA_meth_get_flags+32360
examples/mocks test: 14: 00007FF60F72F36D uv_poll_stop+557
examples/mocks test: 15: 00007FF610540C60 v8::internal::compiler::RepresentationChanger::Uint32OverflowOperatorFor+146416
examples/mocks test: 16: 00007FFF01D37034 BaseThreadInitThunk+20
examples/mocks test: 17: 00007FFF02102651 RtlUserThreadStart+33
examples/mocks test: Failed
undefined
F:\work\projects\quini\GitHub\antfu\vitest-clean\examples\mocks:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @vitest/test-lit@ test: `vitest`
Exit status 134
examples/react-mui test$ vitest
 ```

--- a/packages/vitest/src/node/core.ts
+++ b/packages/vitest/src/node/core.ts
@@ -351,18 +351,31 @@ export class Vitest {
   }
 
   async exit() {
-    const closePromise = this.close()
     let timeout: NodeJS.Timeout
+    let resolveClose: () => void
+    let resolveTimeout: () => void
+    const closePromise = new Promise((resolve, reject) => {
+      resolveClose = () => resolve(true)
+      this.close().then(resolve, reject)
+    })
+    // make sure all premises are resolved before exiting process
+    // https://github.com/nodejs/node/blob/master/src/node_file-inl.h#L160
+    const cleanup = () => {
+      clearTimeout(timeout)
+      resolveTimeout?.()
+      resolveClose?.()
+    }
     const timeoutPromise = new Promise((resolve, reject) => {
+      resolveTimeout = () => resolve(true)
       timeout = setTimeout(() => reject(new Error(`close timed out after ${CLOSE_TIMEOUT}ms`)), CLOSE_TIMEOUT).unref()
     })
     Promise.race([closePromise, timeoutPromise]).then(
       () => {
-        clearTimeout(timeout)
+        cleanup()
         process.exit()
       },
       (err) => {
-        clearTimeout(timeout)
+        cleanup()
         console.error('error during close', err)
         process.exit(1)
       },

--- a/packages/vitest/src/node/core.ts
+++ b/packages/vitest/src/node/core.ts
@@ -358,7 +358,7 @@ export class Vitest {
       resolveClose = () => resolve(true)
       this.close().then(resolve, reject)
     })
-    // make sure all premises are resolved before exiting process
+    // make sure all promises are resolved before exiting process
     // https://github.com/nodejs/node/blob/master/src/node_file-inl.h#L160
     const cleanup = () => {
       clearTimeout(timeout)

